### DISCUSSION TITLE
I can't deploy blue/green smoothly with GKE Ingress.

### DISCUSSION THREAD

**Discussion by kawataki-yoshika** [argoproj/argo-rollouts#1575] [Q&A] [answered]
I installed argo-rollouts in GKE.
In ingress When I promote a blue/green deployment, it returns response code 502 for a few seconds.
Is there any way to fix this?

By the way, the AWS ALB was able to switch without generating any errors.

**Comment by harikrongali** [comment, marked as answer]
Even ALB, that issue is there and we have added a feature https://github.com/argoproj/argo-rollouts/issues/1283 to resolve it.
And for GKE, similar changes are required. And we are looking for contribution on that front.
We will be adding feature in 1.2 https://github.com/argoproj/argo-rollouts/issues/1453 that can fix the issue which will be agnostic to underlying Ingress provider.

**Reply by kawataki-yoshika** [reply]
Thanks for the answer.
I hope 1.2 will be great.

**Comment by soksov** [comment]
This problem still exists

After switching to the new hash to 'active-service', I get several 502 errors within 5-10 seconds with the message: failed_to_pick_backend

**Comment by soksov** [comment]
Hello! I hope this will be useful to someone.

For smooth bg deployment, I decided to use the following algorithm:

when your pods on green are ready, you can patch your “active” k8s service to remove the pod selector with the blue (old) hash, this allows the service to select both replicas, new and old, at the same time. So, after a few seconds, the NEG will update and traffic will be redirected to all pods, after which you will need to scale down the old version, but before that, you must be ensured that you are configured the “Connection draining timeout” in your GCP backend.

All these steps need to be configured during the pre-analysis stage (for example, as a bash script). If everything is successful, argo rollouts will create a selector with a new pod hash for the active service.
